code/src/sixsq/nuvla/server/resources/nuvlabox_playbook.clj

```diff
@@ -0,0 +1,169 @@
+          view-acl (:view-acl nuvlabox-acl)
+          edit-acl (:edit-acl nuvlabox-acl)]
+      (assoc resource
+             :acl (cond-> (assoc (or (:acl resource) (a/default-acl (auth/current-authentication request))) :manage [nuvlabox-id])
+                    (not-empty view-acl) (assoc :view-acl (into [] (distinct (merge view-acl nuvlabox-id))))
+                    (not-empty edit-acl) (assoc :edit-acl edit-acl))))
+    (a/add-acl resource request)))
+
+;;
+;; CRUD operations
+;;
+
+(def add-impl (std-crud/add-fn resource-type collection-acl resource-type))
+
+
+(defmethod crud/add resource-type
+  [{{:keys [parent] :as body} :body :as request}]
+  (some-> parent
+    (db/retrieve request)
+    (a/throw-cannot-edit request))
+  (-> request
+    (update-in [:body] dissoc :output)
+    (add-impl)))
+
+
+(def edit-impl (std-crud/edit-fn resource-type))
+
+
+(defmethod crud/edit resource-type
+  [{:keys [body] :as request}]
+  (if-let [updated-output (:output body)]
+    (edit-impl (assoc request :body (assoc body :output (subs updated-output 0 (min (count updated-output) 1000)))))
+    (edit-impl request)))
+
+
+(def retrieve-impl (std-crud/retrieve-fn resource-type))
+
+
+(defmethod crud/retrieve resource-type
+  [request]
+  (retrieve-impl request))
+
+
+(def delete-impl (std-crud/delete-fn resource-type))
+
+
+(defmethod crud/delete resource-type
+  [request]
+  (delete-impl request))
+
+
+(def query-impl (std-crud/query-fn resource-type collection-acl collection-type))
+
+
+(defmethod crud/query resource-type
+  [request]
+  (query-impl request))
+
+
+;;
+;; Revoke ssh-key action
+;;
+
+
+(defn save-output
+  [{:keys [id output] :as nuvlabox-playbook} new-output]
+  (if new-output
+    (do
+      (try
+        (let [concat-output (str new-output "\n\n" output)
+              request       {:params      {:uuid          (u/id->uuid id)
+                                           :resource-name resource-type}
+                             :body        {:output concat-output}
+                             :nuvla/authn auth/internal-identity}
+              {status :status :as _resp} (crud/edit request)]
+          (if (= 200 status)
+            (log/info "the output from the nuvlabox playbook " id " has been updated")
+            (let [msg (str "cannot update nuvlabox playbook output for " id)]
+              (throw (ex-info msg (r/map-response msg 400 "")))))
+          (r/map-response "playbook output saved successfully" 200))
+        (catch Exception e
+          (or (ex-data e) (throw e)))))
+    (logu/log-and-throw-400 "The provided playbook execution output is empty")))
+
+
+(defmethod crud/do-action [resource-type "save-output"]
+  [{{uuid :uuid} :params body :body :as request}]
+  (try
+    (let [id          (str resource-type "/" uuid)
+          new-output  (:output body)]
+      (-> (db/retrieve id request)
+        (a/throw-cannot-manage request)
+        (save-output new-output)))
```

Comment:
in this case you don't limit the size of the ouput. And it will be bigger later because of concatenation.

Reply:
that's alright cause in the end we'll end up editing the resource, and the string will be truncated

Reply:
there's a test confirming this